src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -82,4 +82,11 @@ default String getAsTextHidePrivate() {
         }
         return builder.toString();
     }
+    default String getPrintableString(Printable... printables) {
+        final StringBuilder builder = new StringBuilder();
+        for(Printable s : printables) {
+            builder.append(s);
```

Comment:
You are trying to append strings here, but s is a Printable instead. You should call getPrintableString method to retrieve the string representation of it.